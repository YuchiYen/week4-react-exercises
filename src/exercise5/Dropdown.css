.Dropdown {}

.Dropdown__header {
    padding: 10px;
    background-color: lightblue;
}

.Dropdown__content {
    padding: 10px;
    background-color: lightcyan;
    border: 1px solid lightblue;
}