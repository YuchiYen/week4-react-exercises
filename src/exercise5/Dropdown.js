import React from 'react';
import PropTypes from 'prop-types';
import './Dropdown.css';

export default class Dropdown extends React.Component {
    static propTypes = {
        header: PropTypes.any.isRequired
    }

    render() {
        const { header, children } = this.props;

        return (
            <div className="Dropdown">
                <div className="Dropdown__header">{header}</div>
                <div className="Dropdown__content">{children}</div>
            </div>
        )
    }
}