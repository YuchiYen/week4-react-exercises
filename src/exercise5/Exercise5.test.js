import React from 'react';
import {shallow} from 'enzyme';
import Exercise5 from './Exercise5';
import Dropdown from './Dropdown';

describe('Exercise5', () => {
    const wrapper = shallow(<Exercise5 />);
    const dropdown = wrapper.find('Dropdown');

    it('should render a Dropdown', () => {
        expect(dropdown).toHaveLength(1);
    });

    it('should have a "header" prop', () => {
        expect(dropdown.prop('header')).toBeDefined();
    })

    it('should have children', () => {
        expect(dropdown.prop('children')).toBeDefined();
    })
});

describe('Dropdown', () => {
    it('should have isOpen in state and a toggleDropdown function', () => {
        const dropdown = shallow(
            <Dropdown header="header text">
                <div className="test-child">test-child</div>
            </Dropdown>
        );

        expect(dropdown.state()).toEqual({
            isOpen: false
        });
        expect(dropdown.instance().toggleDropdown).toBeDefined();
        expect(typeof dropdown.instance().toggleDropdown).toEqual('function');
    });

    it('should not render children by default', () => {
        const dropdown = shallow(
            <Dropdown header="header text">
                <div className="test-child">test-child</div>
            </Dropdown>
        );

        expect(dropdown).toMatchSnapshot();
        expect(dropdown.find('.test-child')).toHaveLength(0);
    });

    it('should render children once the header is toggled', () => {
        const dropdown = shallow(
            <Dropdown header="header text">
                <div className="test-child">test-child</div>
            </Dropdown>
        );
        const header = dropdown.find('.Dropdown__header');
        header.simulate('click');

        expect(dropdown).toMatchSnapshot();
    });
});
