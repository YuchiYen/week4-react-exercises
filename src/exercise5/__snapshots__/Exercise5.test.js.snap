// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should not render children by default 1`] = `
<div
  className="Dropdown"
>
  <div
    className="Dropdown__header"
    onClick={[Function]}
  >
    header text
  </div>
</div>
`;

exports[`Dropdown should render children once the header is toggled 1`] = `
<div
  className="Dropdown"
>
  <div
    className="Dropdown__header"
    onClick={[Function]}
  >
    header text
  </div>
  <div
    className="Dropdown__content"
  >
    <div
      className="test-child"
    >
      test-child
    </div>
  </div>
</div>
`;
